src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/ReclassifyCallbackTest.java

```diff
@@ -0,0 +1,107 @@
+import org.springframework.http.MediaType;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.controllers.CcdCallbackController;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CcdAuthenticator;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CcdAuthenticatorFactory;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.logging.appinsights.SyntheticHeaders;
+
+import java.util.Map;
+
+import static com.google.common.collect.Maps.newHashMap;
+import static java.util.Collections.emptyList;
+import static org.apache.http.HttpHeaders.AUTHORIZATION;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ReclassifyCallbackTest {
+
+    private static final String JOURNEY_CLASSIFICATION_FIELD_NAME = "journeyClassification";
+
+    @Value("${test-url}")
+    private String testUrl;
+
+    @Autowired
+    private CcdAuthenticatorFactory ccdAuthenticatorFactory;
+
+    @Test
+    void should_update_classification_when_request_is_valid() {
+        // given
+        CallbackRequest callbackRequest = prepareValidCallbackRequest();
+
+        // when
+        Response response = callReclassifyEndpoint(callbackRequest);
+
+        // then
+        assertThat(response.statusCode()).isEqualTo(200);
+
+        JsonPath responseJson = response.jsonPath();
+        assertThat(responseJson.getList("errors")).isEmpty();
+        assertThat(responseJson.getList("warnings")).isEmpty();
+
+        Map<String, Object> fieldsInRequest = callbackRequest.getCaseDetails().getData();
+
+
+        Map<String, Object> fieldsInResponse = response.jsonPath().getMap("data");
+        assertThat(fieldsInResponse).isEqualTo(
+            getExpectedFieldsInResponse((fieldsInRequest))
+        );
+    }
+
+    private Response callReclassifyEndpoint(CallbackRequest callbackRequest) {
+        CcdAuthenticator ccdAuthenticator = ccdAuthenticatorFactory.createForJurisdiction("BULKSCAN");
+
+        // when
+        return RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(testUrl)
+            .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+            .header(SyntheticHeaders.SYNTHETIC_TEST_SOURCE, "Bulk Scan Orchestrator Functional test")
+            .header(AUTHORIZATION, ccdAuthenticator.getUserToken())
+            .header(CcdCallbackController.USER_ID, ccdAuthenticator.getUserDetails().getId())
+            .body(callbackRequest)
+            .when()
+            .post("/callback/reclassify-exception-record?ignore-warning=true")
+            .andReturn();
+    }
+
+    private Map<String, Object> getExpectedFieldsInResponse(Map<String, Object> fieldsInRequest) {
+        Map<String, Object> expectedFieldsInResponse = newHashMap(fieldsInRequest);
+        expectedFieldsInResponse.put("journeyClassification", "SUPPLEMENTARY_EVIDENCE_WITH_OCR");
```

Comment:
are there any constant for both of those?

Reply:
I've created a constant in this test class for journeyClassification (it's used in more than one place), just forgot to use it.

As for `SUPPLEMENTARY_EVIDENCE_WITH_OCR `, a constant exists only in application code (test values have to be independent from application values). I can add a constant in this test class, but the value is only used once, anyway.